src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs

```diff
@@ -6,43 +6,104 @@
+            while (receiver.State == WebSocketState.Open && sender.State == WebSocketState.Open) {
+                if (ct.IsCancellationRequested) {
+                    Task.WhenAll(
+                        receiver.CloseAsync(WebSocketCloseStatus.NormalClosure, "Normal closure", CancellationToken.None),
+                        sender.CloseAsync(WebSocketCloseStatus.NormalClosure, "Normal closure", CancellationToken.None))
+                        .DoNotWait();
+                    return;
+                }
+
+                WebSocketReceiveResult result = await receiver.ReceiveAsync(receiveBuffer, ct);
+
+                byte[] data = null;
+                using (MemoryStream ms = new MemoryStream()) {
+                    await ms.WriteAsync(receiveBuffer.Array, receiveBuffer.Offset, result.Count);
+                    await ms.FlushAsync();
+                    data = ms.ToArray();
+                }
+
+                ArraySegment<byte> sendBuffer = new ArraySegment<byte>(data);
+                await sender.SendAsync(sendBuffer, result.MessageType, result.EndOfMessage, ct);
+
+                if (result.MessageType == WebSocketMessageType.Close) {
+                    await receiver.CloseAsync(WebSocketCloseStatus.NormalClosure, "Normal closure", ct);
+                }
+            }
+        }
+
+        private Task DoWebSocketWorkAsync(WebSocket localWebSocket, WebSocket remoteWebSocket, CancellationToken ct) {
+            return Task.WhenAll(
+                DoWebSocketReceiveSendAsync(localWebSocket, remoteWebSocket, ct),
+                DoWebSocketReceiveSendAsync(remoteWebSocket, localWebSocket, ct));
+        }
+
         public async Task GetResponseAsync(HttpListenerContext context, string localBaseUrl, string remoteBaseUrl, CancellationToken ct) {
-            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(PostUri);
+            string postUri = null;
+            if (context.Request.IsWebSocketRequest) {
+                UriBuilder ub = new UriBuilder(PostUri) { Scheme = "ws" };
+                postUri = ub.Uri.ToString();
+            } else {
+                postUri = PostUri.ToString();
+            }
+
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(postUri);
             request.Method = context.Request.HttpMethod;
-            SetRequestHeaders(request, context.Request.Headers, localBaseUrl, remoteBaseUrl);
+
+            if (!context.Request.IsWebSocketRequest) {
+                SetRequestHeaders(request, context.Request.Headers, localBaseUrl, remoteBaseUrl);
+            }
 
             // Add RTVS headers
             request.Headers.Add(CustomHttpHeaders.RTVSRequestedURL, GetRemoteUrl(context.Request.Url, remoteBaseUrl));
 
-            if(context.Request.InputStream.CanSeek && context.Request.InputStream.Length > 0) {
+            if (context.Request.InputStream.CanSeek && context.Request.InputStream.Length > 0) {
```

Comment:
Kinda surprised that `CanSeek` is true here - I would normally expect it to not be for a request, since it's just reading from a socket ultimately.

Why is it needed? It doesn't look like there's any seeking below...